scripts/firedrake-install

```diff
@@ -86,6 +86,8 @@ honoured.""",
                         help="Minimise the set of petsc dependencies installed. This creates faster build times (useful for build testing).")
     parser.add_argument("--honour-petsc-dir", "--honour_petsc_dir", action="store_true",
                         help="Usually it is best to let Firedrake build its own PETSc. If you wish to use another PETSc, set PETSC_DIR and pass this option.")
+    parser.add_argument("--honour-pythonpath", "--honour_pythonpath", action="store_true",
+                        help="Pointing to external Python packages is usually a user error. Set this option if you know that you want PYTHONPATH set.")
```

Comment:
Because of the hacky way that we save configuration. For install, it's fine to just set args.honour_pythonpath. For upgrade you have to take into account a possible previous saved value.